packages/safe-kit/src/types.ts

```diff
@@ -0,0 +1,43 @@
+import Safe, { SafeProvider } from '@safe-global/protocol-kit'
+import { TransactionBase, TransactionOptions } from '@safe-global/safe-core-sdk-types'
+
+export type SafeConfig = {
+  owners: string[]
+  threshold: number
+  saltNonce?: string
+}
+
+export type ExistingSafeKitConfig = {
+  safeAddress?: string
+  safeOptions?: never
+}
+
+export type PredictedSafeKitConfig = {
+  safeAddress?: never
+  safeOptions?: SafeConfig
+}
+
+export type SafeKitRootConfig = {
+  provider: SafeProvider['provider']
+  signer: SafeProvider['signer']
+}
+
+export type SafeKitConfig = SafeKitRootConfig & (ExistingSafeKitConfig | PredictedSafeKitConfig)
+
+export type SafeClientTransactionResult = {
+  chain: {
+    hash?: string | undefined
+  }
+  safeServices?: {
+    safeTxHash: string | undefined
+  }
+}
+
+export type SafeClient = {
```

Comment:
Do we need this type? Can we not use the `SafeAccountCient` class as type directly? Or what was the idea behind defining the type separately and making the class implement it?

Reply:
hehe, no, typical thing you start implementing one way and end with other. Good suggestion

Reply:
changed